components/blitz/src/omero/gateway/facility/TablesFacilityHelper.java

```diff
@@ -225,6 +226,8 @@ void parseData(Data data, TableDataColumn[] header) {
                 for (int j = 0; j < nRows; j++) {
                     MaskData md = new MaskData(mc.x[j], mc.y[j], mc.w[j],
                             mc.h[j], mc.bytes[j]);
+                    md.setZ(mc.theZ[j]);
```

Comment:
Will ``theZ`` and ``theT`` always be filled?

Reply:
I'd expect that both are always arrays of `nRows` size, but you're right, I guess the values could be `-1` if `z` and `t` are not explicitly set. Which is actually a bug in `ShapeData` (`getZ` and `getT` should return `0` not `-1` in that case).  Follow-up PR maybe?

Reply:
don't modify in this PR
Changing the getter and setter should be done in another PR